rfq/manager.go

```diff
@@ -1212,6 +1215,61 @@ func (m *Manager) RfqChannel(ctx context.Context,
 	return balancesMap, nil
 }
 
+// InboundPolicyFetcher is a helper that fetches the inbound policy of a channel
+// based on its chanID.
+type InboundPolicyFetcher func(ctx context.Context, chanID uint64,
+	remotePubStr string) (*lnrpc.RoutingPolicy, error)
+
+// RfqToHopHint creates the hop hint representation which encapsulates certain
+// quote information along with some other data required by the payment to
+// succeed.
+func (m *Manager) RfqToHopHint(ctx context.Context,
+	policyFetcher InboundPolicyFetcher, channelID uint64,
+	peerPubKey route.Vertex, quote *rfqrpc.PeerAcceptedBuyQuote,
+	hold bool) (*lnrpc.HopHint, []zpay32.HopHint, error) {
+
+	inboundPolicy, err := policyFetcher(ctx, channelID, peerPubKey.String())
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to get inbound channel "+
+			"policy for channel with ID %d: %w", channelID, err)
+	}
+
+	if hold {
+		peerPub, err := btcec.ParsePubKey(peerPubKey[:])
+		if err != nil {
+			return nil, nil, fmt.Errorf("error parsing peer "+
+				"pubkey: %w", err)
+		}
+		hopHint := []zpay32.HopHint{
+			{
+				NodeID:      peerPub,
+				ChannelID:   quote.Scid,
+				FeeBaseMSat: uint32(inboundPolicy.FeeBaseMsat),
+				FeeProportionalMillionths: uint32(
+					inboundPolicy.FeeRateMilliMsat,
+				),
+				CLTVExpiryDelta: uint16(
+					inboundPolicy.TimeLockDelta,
+				),
+			},
+		}
+
+		return nil, hopHint, nil
```

Comment:
As we only ever return a single hop hint here (right now), seems we can just return it directly.

Reply:
If we return that directly, then maybe we can make a thin wrapper struct which can take our "rfq" hop hint, and map it into either the `zpay32` version (just return the embedded version), or map it into the RPC version.